cmd/gpu-kubelet-plugin/driver.go

```diff
@@ -177,6 +195,112 @@ func (d *driver) nodeUnprepareResource(ctx context.Context, claimNs kubeletplugi
 	return nil
 }
 
+func (d *driver) deviceHealthEvents(ctx context.Context, nodeName string) {
+	klog.Info("Processing device health notifications")
+	for {
+		select {
+		case <-ctx.Done():
+			klog.V(6).Info("Stop processing device health notifications")
+			return
+		case device, ok := <-d.deviceHealthMonitor.Unhealthy():
+			if !ok {
+				klog.V(6).Info("Health monitor channel closed")
+				return
+			}
+
+			uuid := device.GetUUID()
+			klog.Warningf("Received unhealthy notification for device: %s", uuid)
+
+			// Mark device as unhealthy.
+			d.state.MarkDeviceUnhealthy(device)
+
+			// Update allocated device health status in a given claim
+			result, claimUID, err := d.findDeviceResultAndClaimUID(uuid)
+			if err != nil {
+				klog.Errorf("Device %s is unhealthy, but no associated claim was found: %v", uuid, err)
+				continue
+			}
+
+			claim, err := d.findClaimByUID(ctx, types.UID(claimUID))
+			if err != nil {
+				klog.Errorf("Failed to find ResourceClaim object for UID %s: %v", claimUID, err)
+				continue
+			}
+
+			ds := d.state.buildDeviceStatusApply(result, device.Health)
+			if err := d.state.applyClaimDeviceStatuses(ctx, claim.Namespace, claim.Name, ds); err != nil {
+				klog.Errorf("Failed to update status for claim %s/%s: %v", claim.Namespace, claim.Name, err)
+			} else {
+				klog.V(6).Infof("applied unhealthy device status to claim %s/%s", claim.Namespace, claim.Name)
+			}
```

Comment:
I don't think we should just give up here. Its likely we may have a conflict when writing and we need to try again. Instead we should push a task to the workqueue that will keep retrying until it succceeds.